go/marshal/encode_test.go

```diff
@@ -694,6 +694,62 @@ func TestEncodeOriginal(t *testing.T) {
 	assert.True(out.Type().Equals(st))
 }
 
+type MarshalOptionsTestT struct {
+	Abc  int
+	opts MarshalOptions `noms:"-"`
+}
+
+func (t MarshalOptionsTestT) GetMarshalOptions() MarshalOptions {
+	return t.opts
+}
+
+func TestEncodeMarshalOptionsName(t *testing.T) {
+	assert := assert.New(t)
+
+	s := MarshalOptionsTestT{
+		Abc: 42,
+		opts: MarshalOptions{
+			Name: "",
+		},
+	}
+	v, err := Marshal(s)
+	assert.NoError(err)
+	assert.True(types.NewStruct("MarshalOptionsTestT", types.StructData{
```

Comment:
I find this API strange. I would want `Name: ""` to result on a Noms struct without a name (`""`).

If we want a way to use the Go name I think that should be done with an extra option and not with the empty string.